Avoid copying over data in the write_vio into response_buffer

masaori335 opened the pull request:
At `Http2Stream::update_write_request()`, data in the `write_vio` is copied to `response_buffer` and consumed immediately . But we don't need it.
Also, this could be fix of #2530. Because the `response_buffer` is removed.

----

Notes: This PR includes commits of #2837. Please look at last commit. I'll rebase after #2837 is merged.

bryancall commented:
This PR can remove the need for checking the size of the response buffer in #2534.  However, #2534 also checks the size of the chunked_handler buffer.

masaori335 commented:
@bryancall I rebased this PR on latest master (includes #2534) and removed only `response_buffer` size check.
It'd be great if `ChunkedHandler` or `HttpTunnel` have `dechunked_buffer` size check. But it looks out of scope of this PR.

masaori335 commented:
@zwoop I'll rebase this PR when #2837 is merged. The first two commits will disappear.

masaori335 commented:
[approve ci autest]

masaori335 commented:
[approve ci CentOS]
[approve ci Ubuntu]

masaori335 commented:
Rebased on latest master (c802ea2f95af2360520fc75d57266586271e0196)

masaori335 commented:
Ah, I rebased on broken master. Rebased latest master again.

masaori335 commented:
[approve ci autest]

masaori335 commented:
[approve ci autest]

zwoop commented:
This doesn't cherry-pick cleanly, @masaori335 Can you make a PR against the 7.1.x branch please?

zwoop commented:
This breaks larger downloads (they stall). I *think* anything > 1MB will fail over H2 now with this, I backed out this change from Docs, and it makes things work again.
